packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -113,14 +132,61 @@ const useDesktopPlayer = ({
     canvas.width = spec.width;
     canvas.height = spec.height;
 
-    setShowCanvas(true);
+    setAttempt({ status: 'success' });
+  };
+
+  useEffect(() => {
+    if (playerClient) {
+      playerClient.addListener(PlayerClientEvent.SESSION_END, () => {
+        setAttempt({ status: '' });
+      });
+
+      playerClient.addListener(
+        PlayerClientEvent.PLAYBACK_ERROR,
+        (err: Error) => {
+          setAttempt({
+            status: 'failed',
+            statusText: 'playback error: ' + err.message,
```

Comment:
"playback error" isn't quite clear enough to me. Suggestion:

```suggestion
            statusText: `There was an error while playing this session: ${err.message}`,
```